processing/src/main/java/org/apache/carbondata/processing/loading/iterator/CarbonOutputIteratorWrapper.java

```diff
@@ -45,17 +45,21 @@
 
   private RowBatch readBatch;
 
+  private static Object lockObject = new Object();
+
   private ArrayBlockingQueue<RowBatch> queue = new ArrayBlockingQueue<>(10);
 
   public void write(Object[] row) throws InterruptedException {
     if (close) {
       // already might be closed forcefully
       return;
     }
-    if (!loadBatch.addRow(row)) {
-      loadBatch.readyRead();
-      queue.put(loadBatch);
-      loadBatch = new RowBatch(batchSize);
+    synchronized (lockObject) {
```

Comment:
why not use `this`

Reply:
yes, can use this as this is object level synchronization

Reply:
if used 'this', findbugs thinks I need to synchronize hasNext() and close() method also. It says inconsistent synchronization.  Hence to avoid that this variable.

Reply:
I feel that the loadBatch should be the parameter of this function, so the caller can accumulate the rows in a batch in its own thread, then call this write function to add to the shared queue.